### ISSUE TITLE
[RFE] Improve RKE1 UX when registering a node with a duplicate hostname/ip as an existing node

### ISSUE THREAD

**Issue by jakefhyde** [rancher/rancher#39780]
**Is your feature request related to a problem? Please describe.**

For both node driver and custom clusters, when a new node attempts to join the cluster with an ip address or hostname that matches an existing node, RKE1 will attempt to prevent the node from joining. In some cases, however, this detection fails (if for example, the `HostnameOverride` is not set, the kubelet may attempt to register with an existing hostname, such as localhost). 

**Describe the solution you'd like**

Prevent RKE1 from attempting to bring up the kubelet on the node, and instead warn that the hostname is a duplicate (or impossible, if the hostname is localhost). Additionally, it would also be an improvement to the UI to denote for custom clusters that the selected ip address or hostname is duplicated.

**Comment by annablender** [comment]
**Design discussion summary**

Our team agreed that this issue is low priority and needs additional discussion around ROI/LOE vs. Support allowing this _known issue_ to slip through the cracks.

**LOE** - implementing this as an error returned from the RKE1 CLI is easy. How it's presented in Rancher without appearing to break cluster provisioning is much harder. This is an issue for both imported and custom RKE1 clusters, which encompasses a lot of customer cases and we may not be able to fix this to even cover 90% of cases. Plus, it should be basic to figure out if a customer is having this issue because only 1 node will register and RKE1 will prevent subsequent nodes from registering so ROI may be too low.

**Support** - @jakefhyde This known issue causes a ton of customer escalations and enhancing the error handling would lower the number of escalations, but it also may not matter because Support is sending issues to dev engineering without checking for this bug.

**In other words**, why fix this in _our code_ if the issue is very difficult to implement and can be fixed by updating docs and improving Support's knowledge on known issues?

**Comment by snasovich** [comment]
https://github.com/rancher/rancher-docs/pull/729 got merged so the documentation update is covered.
However, it's unlikely we will get to actually improve UI/UX beyond the documentation so closing it as "Won't Do". If the priorities change it can always be reopened.
